.github/workflows/codeql.yml

```diff
@@ -0,0 +1,40 @@
+name: "CodeQL Advanced"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  schedule:
+    - cron: '35 21 * * 4'
```

Comment:
Is this intended to re-assess the code quality in case we don't merge anything for a week? If we run the analysis on every PR or merge we should be fine, no?